dbms/src/Storages/DeltaMerge/Index/LocalIndexInfo.cpp

```diff
@@ -302,11 +219,11 @@ LocalIndexInfosChangeset generateLocalIndexInfos(
 
     // only return the newly dropped index with index_id > EmptyIndexID
     std::vector<IndexID> dropped_indexes;
-    for (const auto & i : newly_dropped)
+    for (const auto & idx_id : newly_dropped)
     {
-        if (i.index_id <= EmptyIndexID)
+        if (idx_id <= EmptyIndexID)
             continue;
-        dropped_indexes.emplace_back(i.index_id);
+        dropped_indexes.emplace_back(idx_id);
     }
     return LocalIndexInfosChangeset{
         .new_local_index_infos = new_index_infos,
```

Comment:
As all index_ids are get from the index_info now, we can directly return `newly_dropped`

Reply:
fixed